mllib/src/main/scala/org/apache/spark/ml/recommendation/TopByKeyAggregator.scala

```diff
@@ -57,3 +57,57 @@ private[recommendation] class TopByKeyAggregator[K1: TypeTag, K2: TypeTag, V: Ty
 
   override def outputEncoder: Encoder[Array[(K2, V)]] = ExpressionEncoder[Array[(K2, V)]]()
 }
+
+
+/**
+ * Works on rows of the form (ScrId, DstIds, Scores).
+ * Finds the top `num` DstIds and Scores.
+ */
+private[recommendation] class TopKArrayAggregator(num: Int)
```

Comment:
I don't love the copy-paste - are the other uses of the class above able to use your new idea?

Reply:
since I just find that aggregating on array bring little improvement, so I will remove this class